### ISSUE TITLE
Rust highlighting doesn't support current inner-attribute syntax

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#679]
(Original issue 972 created by eridius on 2014-04-09T00:03:54.385424+00:00)

The Rust language changed its inner-attribut syntax from `#[attr];` to `#![attr]`. Pygments doesn't seem to handle this and marks up the leading # in `#![attr]` as an error.

I realize that Rust is still in development and having highlighting get a bit out of date is to be expected, but the obvious error markup for `#![attr]` is rather unfortunate. Most language changes don't get flagged as errors by syntax highlighters.

**Comment by Anteru** [comment]
(Original comment by tshatch on 2014-04-15T17:20:40.374442+00:00)

Please see Pull Request 320, let me know if that's valid.
https://bitbucket.org/birkenfeld/pygments-main/pull-request/320/include-new-attribute-syntax-for-rust/diff

**Comment by Anteru** [comment]
(Original comment by eridius on 2014-04-17T13:15:06.458358+00:00)

The semicolon for inner attributes should be removed. It's still valid at the moment (though not in conjunction with the "!") but it will be going away at some point.

**Comment by Anteru** [comment]
(Original comment by tshatch on 2014-04-23T19:51:41.981117+00:00)

I don't speak rust; can you give me a syntactically valid example for what "inner" and "outer" mean?

**Comment by Anteru** [comment]
(Original comment by eridius on 2014-04-24T07:36:00.425252+00:00)

An "outer" attribute is one that modifies the item that directly follows it in the source. An item may be a module, a type definition, a function, etc. An "inner" attribute is one that modifies the item that the attribute is contained within (and the inner attribute must be the first thing inside of the item).

For example, the following demonstrates an outer attribute for a function:

```rust
#[inline]
pub fn foo(x: int) {
    println!("foo: {}", x);
}
```

Whereas the following is the same function using an inner attribute:

```rust
pub fn foo(x: int) {
    #![inline]
    println!("foo: {}", x);
}
```

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-10-07T13:25:25.138092+00:00)

Closes #972: accept "inner attribute" syntax for Rust.

→ <<cset 43c74f79c17b>>
